### PR TITLE
[AWS] Add support for global regions in AWS

### PR DESCRIPTION
**Pull request by Michaelvll**
This PR adds experimental supports for global regions for AWS.

We use US regions by default. The advanced users who would like to try the global regions will need to create a file: `~/.sky/config`
```
catalog:
  aws:
    preferred_areas: all # can be changed to ["us", "eu", "ap"]
```
I am planning to add a `sky config` CLI that can let the user to set some configurations for SkyPilot. After that, we will not let the user know about the existence of `~/.sky/config`. 

Please let me know how do you feel about this design @concretevitamin, @romilbhardwaj, @WoosukKwon and @ all?

Tested:
- [x] `sky launch --cloud aws --region ap-south-1`

### CODE REVIEW

File: sky/clouds/aws.py [hunk 1]
```diff
@@ -127,9 +127,27 @@ def get_default_ami(cls, region_name: str, instance_type: str) -> str:
         amis = {
             'us-east-1': 'ami-0729d913a335efca7',
             'us-east-2': 'ami-070f4af81c19b41bf',
-            # This AMI is 20210623 as aws does not provide a newer one.
-            'us-west-1': 'ami-0b3c34d643904a734',
+            'us-west-1': 'ami-0365e546798d80d88',
```
Comment (Michaelvll) [inline sky/clouds/aws.py:130]: This region does have the `20220308` version of the AMI. It should be supported.

**Comment by romilbhardwaj** [comment]
I think we really need a `~/.sky/config` file to store user specific configuration (e.g. custom flags for storage mounting), so this a great step! I'm not fully sure if we need a `sky config` cli command. I feel we can ask users to directly edit the `~/.sky/config` file when needed.

Also curious - why don't we enable all geographic regions by default? Because of failover/quota?

File: sky/clouds/service_catalog/common.py [hunk 2]
```diff
@@ -46,15 +48,90 @@ def get_catalog_path(filename: str) -> str:
...
+        The cache path.
+    """
+    assert filename.endswith('.csv'), filename
+    new_name = filename.split('.csv')[0] + '.filtered.'
+    cached_catalog_path = get_catalog_path(new_name)
+    cached_catalog_path += f'{"_".join(area_filter)}.csv'
+    return cached_catalog_path
+
+
+def _read_catalog_config() -> Optional[Dict[str, Dict[str, str]]]:
```
Comment (romilbhardwaj) [inline sky/clouds/service_catalog/common.py:68]: Nice, eventually we should converge on something like a `SkyConfig` class which exposes methods to read the `config` file.

**Comment by concretevitamin** [comment]
At a high level I'd like to propose sticking with not introducing `~/.sky/config` or `sky config` for as long as possible. Main reason is keeping the launch path stateless makes the system simpler to reason about and use, for both developers and users. 

For example, we can imagine a potential interface that changes the current field in a task YAML
```
region: us-west1
```
to any of these:
```
region: [us-west1, us-east1]  # Need validation logic in case zone is specified.
region: [us, eu]  # Need validation logic in case zone is specified.
locations: [us, eu]  # Need validation logic in case region/zone are specified.
```
Removing the global config state also means each task can easily specify different locations.

At some point a very compelling reason for a separate config file may show up, but I think the reasons so far do not definitely call for it (for all continents, see above; for storage mount options, we are adding / have added the flags as defaults).

**Comment by romilbhardwaj** [comment]
I think the decision depends on this question:

> why don't we enable all geographic regions by default? Because of failover/quota?

* If there's no reason, we should enable all regions. Then we don't need this option to be exposed at all.
* If the reason is to limit SkyPilot (provisioning, failover) to specific regions, then it is likely better UX to specify my geographic region preferences once in a config file rather than annotating `geo_region: [eu]` in every YAML I use and specifying `--geo-region [eu]` in every cpunode command. 
  * This solution is probably cumbersome when the user has per-task geo_region constraints, but I think that might be a rare case.   
  * That said, I'm not sure why these preferences (either globally or per-task) would exist in the first place. Data sovereignty? 
* If it is some other reason, then we should discuss it.

Orthogonal to geo_regions, I feel many SkyPilot components sit in a very rich tradeoff space (e.g. Storage - `--implicit-dirs` has likely increased gcs costs but provided convenience, our current stat/type cache timeouts for storage mounts prioritize consistency over cost, other storage mounting flags we don't touch). While we try our best to pick sensible defaults, there's no way we can optimize for each user without yielding them control over these tradeoff knobs. If we expose these global controls through a `config` file, it 1. keeps our YAML simple 2. Allows users to dial the knobs to whatever works best for them rather than having them wait for us to implement/support it. Of course, only power users use this `config` file, and most of our users never even see it.

**Comment by Michaelvll** [comment]
After discussing offline, we decided to defer this for now due to failing to reach a consensus. For users who would like to use global regions can run the following command to let SkyPilot use global regions.
```
cd ~/.sky/catalogs/v4
# Fetch all regions for AWS
python -m sky.clouds.service_catalog.data_fetchers.fetch_aws --all-regions
```
https://skypilot.readthedocs.io/en/latest/reference/faq.html#advanced-how-to-make-skypilot-use-all-global-regions

**Comment by Michaelvll** [comment]
This has been adopted to #1449, closing for now.

**Pull request closed without merging**